k8s-plugin-keel/src/main/kotlin/com/amazon/spinnaker/keel/k8s/verificationEvaluator/K8sJobEvaluator.kt

```diff
@@ -173,8 +173,13 @@ class K8sJobEvaluator(
         }
         // apply verifyWith specific labels
         labels[VERIFICATION_ENVIRONMENT_LABEL] = environment
-        labels[VERIFICATION_ARTIFACT_LABEL] = artifactVersion
+        labels[VERIFICATION_ARTIFACT_LABEL] = validK8sName(artifactVersion)
 
         metadata[LABELS] = labels
     }
+
+    private fun validK8sName(k8sName: String): String {
+        val re = Regex(K8S_INVALID_CHARACTER_REGEX)
```

Comment:
This regex should be moved to this class' private property since it's static and better to compile regex once.

Reply:
good point, just pushed an update.